specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/preview/2022-09-01-preview/vaults.json

```diff
@@ -0,0 +1,2128 @@
+            "$ref": "#/definitions/ClientDiscoveryValueForSingleApi"
+          }
+        },
+        "nextLink": {
+          "description": "Link to the next chunk of the response",
+          "type": "string"
+        }
+      }
+    },
+    "ClientDiscoveryValueForSingleApi": {
+      "description": "Available operation details.",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "Name of the Operation.",
+          "type": "string"
+        },
+        "display": {
+          "$ref": "#/definitions/ClientDiscoveryDisplay",
+          "description": "Contains the localized display information for this particular operation"
+        },
+        "origin": {
+          "description": "The intended executor of the operation;governs the display of the operation in the RBAC UX and the audit logs UX",
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/ClientDiscoveryForProperties",
+          "description": "ShoeBox properties for the given operation."
+        }
+      }
+    },
+    "ImmutabilitySettings": {
+      "description": "Immutability Settings of vault",
+      "type": "object",
+      "properties": {
+        "state": {
+          "enum": [
+            "Disabled",
+            "Unlocked",
+            "Locked"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ImmutabilityState",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "MonitoringSettings": {
+      "description": "Monitoring Settings of the vault",
+      "type": "object",
+      "properties": {
+        "azureMonitorAlertSettings": {
+          "description": "Settings for Azure Monitor based alerts",
+          "$ref": "#/definitions/AzureMonitorAlertSettings"
+        },
+        "classicAlertSettings": {
+          "description": "Settings for classic alerts",
+          "$ref": "#/definitions/ClassicAlertSettings"
+        }
+      }
+    },
+    "Resource": {
+      "description": "ARM Resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "description": "Resource Id represents the complete path to the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "name": {
+          "description": "Resource name associated with the resource.",
+          "type": "string",
+          "readOnly": true
+        },
+        "type": {
+          "description": "Resource type represents the complete path of the form Namespace/ResourceType/ResourceType/...",
+          "type": "string",
+          "readOnly": true
+        },
+        "etag": {
+          "description": "Optional ETag.",
+          "type": "string"
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "SecuritySettings": {
+      "description": "Security Settings of the vault",
+      "type": "object",
+      "properties": {
+        "immutabilitySettings": {
```

Comment:
>immutabilitySettings [](http://example.com/codeflow?start=9&length=20)

I think "mutability" is a better name for this property.

Reply:
This name is similar to what is used by storage (Blob) for the property having same effect

[blob swagger link](https://github.com/pratimaupadhyay02/azure-rest-api-specs/blob/be4eb616b6d5008245492ba6d1eb9ab62a75e87d/specification/storage/data-plane/Microsoft.BlobStorage/preview/2020-10-02/blob.json#L9774)

Reply:
These are the settings to make resource Immutable. Resource is considered mutable if these settings are not enabled.

Reply:
These settings control the mutability of the resource. The general term is mutability which can be set to mutable or immutable. It's cleanest to name the properties and values without unnecessary negation built into the name.

This
{
 . . .
  securitySettings: {
    immutabilitySettings: {
      state: { Disabled, Locked, Unlocked }
    }
  }
}

is wordy and confusing (what's the difference between Disabled and Unlocked?). Something more like this:
{
  mutability: { Mutable, Immutable }
}

is simpler and easier to understand.

I won't block ARM approval on this, just want you to reconsider.

Reply:
**Disable** means Immutability is not enabled yet i.e resource is mutable
**Unlocked** means Immutability is enabled (resource is not mutable) but not yet locked that means it can be reveted from this state.
**Locked** means Immutability is Locked (resource is not mutable) forever and cannot be reverted at all.

I agree with you we should avoid any negation in Names but all existing resources created till now are mutable by default, now we are exposing feature to make those Immutable. Hence kept name as Immutability Settings and same is used for other resources like RecoveryServiceVaults and Storage resources. As per industry standard also, Immutability is being used in ransomware protection solutions.